RecoLocalCalo/EcalRecProducers/plugins/EcalUncalibRecHitPhase2WeightsAlgoGPU.cu

```diff
@@ -0,0 +1,35 @@
+#include "DataFormats/EcalDigi/interface/EcalDataFrame_Ph2.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+
+#include "EcalUncalibRecHitPhase2WeightsKernels.h"
+#include "EcalUncalibRecHitPhase2WeightsAlgoGPU.h"
+
+namespace ecal {
+  namespace weights {
+
+    void entryPoint(ecal::DigisCollection<calo::common::DevStoragePolicy> const& digis,
+                    EventOutputDataGPU& eventOutputGPU,
+                    cms::cuda::device::unique_ptr<double[]>& weights_d,
+                    cudaStream_t cudaStream) {
+      unsigned int totalChannels = digis.size;
+      // 64 threads per block best occupancy from Nsight compute profiler
+      unsigned int nchannels_per_block = 64;
+      unsigned int threads_1d = nchannels_per_block;
+      unsigned int blocks_1d = (totalChannels / threads_1d) + 1;
+      // shared bytes from size of gain array, weight constants, digi samples per block, uncalib rechits amplitudes per block
+      int shared_bytes = EcalDataFrame_Ph2::MAXSAMPLES * sizeof(double) +
```

Comment:
Have the gain array bytes been removed already now that the ecalPh2::gains is used directly? They are still mentioned in the comment.